db/migrate/20231214151708_add_devise_to_users.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class AddDeviseToUsers < ActiveRecord::Migration[7.1]
+  def self.up
+    change_table :users do |t|
+      ## Database authenticatable
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
```

Comment:
-  You have done great so far :+1: :+1: But there are a few requirements you need to meet before moving to the next milestone. :+1: 
- [ ] When I run `rails db:migrate`, I Got these errors shown in the screenshot below. The error points out the email does not exist. Looking at your `AddDeviseToUsers` migration file, it seems you are missing the `email` column which is a required column. I suggest you update this migration file to include the email column. You can add the email column code above `line 7`. 
```ruby
t.string :email,              null: false, default: ""
```

- - Error screenshot:
![Selection_068](https://github.com/Chimoneg27/Rails-Blog-App/assets/107632780/671c9f5d-4d18-4edb-9072-184c12c2a4fb)

Reply:
You don't need to run these. It works fine on my side. That process took me hours to fix.